decimal64math.go

```diff
@@ -124,6 +124,52 @@ func (d Decimal64) Sqrt() Decimal64 {
 	exp, significand = renormalize(exp/2-8, sqrt)
 	return newFromParts(sign, exp, significand)
 }
+func (dp *decParts) round128to64() discardedDigit {
+	var rndStatus discardedDigit
+	if dp.significand.numDecimalDigits() > 16 {
+		var remainder uint64
+		expDiff := dp.significand.numDecimalDigits() - 16
+		dp.exp += expDiff
+		dp.significand, remainder = dp.significand.divrem64(powersOf10[expDiff])
+		rndStatus = roundStatus(remainder, 0, expDiff)
+	}
+	// dp.significand.lo = dp.significand.lo
+	return rndStatus
+}
+
+func (dp *decParts) Add128(ep *decParts) decParts {
+	targetExp := ep.exp - dp.exp
+	var ans decParts
+	if (ep.significand != uint128T{0, 0}) {
+		if targetExp < 0 {
+			dp.significand = dp.significand.mul(powerOfTen128(targetExp))
+			dp.exp += targetExp
+		} else if targetExp > 0 {
+			ep.significand = ep.significand.mul(powerOfTen128(targetExp))
+			ep.exp -= (targetExp)
```

Comment:
for line 146, on my local copy you changed it to `dp.exp -= -targetExp`, for consistency line 149 would become `ep.exp -= targetExp`